File: GastosResidenciais.Backend/Domain/Entities/Transacao.cs
using System.ComponentModel.DataAnnotations;
using GastosResidenciais.Backend.Domain.Enums;

namespace GastosResidenciais.Backend.Domain.Entities
{

    public class Transacao
    {
        public int Id { get; set; }

        [Required]
        [MaxLength(400)]
        public string Descricao { get; set; } = string.Empty;

        [Range(0.01, double.MaxValue)]
        public decimal Valor { get; set; }

        public TipoTransacao Tipo { get; set; }

        public int PessoaId { get; set; }
        public Pessoa Pessoa { get; set; } = null!;

        public int CategoriaId { get; set; }
        public Categoria Categoria { get; set; } = null!;

         [Required]
        public DateTime Data { get; set; }
    }
}

File: gastos-residenciais-front/src/types/Transacao.ts
export interface Categoria {
  id: number;
  descricao: string;
}

export interface Pessoa {
  id: number;
  nome: string;
  idade: number;
}

export interface Transacao {
  id: number;
  descricao: string;
  valor: number;
  tipo: 0 | 1; // 0 = Receita, 1 = Despesa
  categoria: Categoria;
  pessoa?: Pessoa;
  data: string;
}

export interface CreateTransacaoPayload {
  descricao: string;
  valor: number;
  tipo: 0 | 1;
  categoriaId: number;
  pessoaId: number;
  data: string;
}